### PR TITLE
feat: add support for Time based OTP

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ Part of https://github.com/elastic/synthetics-dev/issues/367
+ Add first class MFA support with a prebundled TOTP library `otpauth`. Its actively maintained and works on all runtimes.  
+ Synthetics library exposes a `mfa.totp` function that can be used to generate a One time token during the journey run. 
```ts
import { journey, step, mfa} from '@elastic/synthetics';

journey('2FA', ({ page }) => {
  step('Login using 2FA', async () => {
    // login using username and pass and go to 2FA in next page
    const token = mfa.token("NB2W45DFOIZA");
    await page.getByPlaceholder("token-input").fill(token)
  });
});
```
+ Users can also configure it for Inline and other Project based journeys by specifying these MFA secrets from params which would allow for generating the token at runtime instead of generating from the UI which would cause authentication issues as journeys would run past the period the token is valid for. 

```ts
    const token = mfa.token(params.MFA_GH_SECRET);
```

### CODE REVIEW

**Review by shahzad31 — APPROVED** [review]
LGTM !!

Tested with https://otplib.yeojz.dev/ 

and it seems to work as expected !!

File: src/core/mfa.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+   */
+  digits?: number
+  /**
+   * Validity period in seconds for the token.
+   * @default 30
+   */
+  period?: number
+};
+
+export function totp(secret?: string, options: TOTPOptions = {}) {
```
Comment (shahzad31) [inline src/core/mfa.ts:63]: how come secret is optional? isn't that always required?

Comment (vigneshshanmugam) [reply src/core/mfa.ts:63]: No, if we dont pass anything, then it gets auto generated. Just a connivence if all we want is a 6 digit code.

**Review by shahzad31 — APPROVED** [review]
LGTM !!

File: src/core/mfa.ts (hunk 1 above)
Comment (emilioalvap) [reply src/core/mfa.ts:63]: nit: should this make `issuerInLabel: true` explicit here? Instead of relying on the library default

Comment (vigneshshanmugam) [reply src/core/mfa.ts:63]: We could do it though, however different platforms GH / Cloudflare differs here. GH explicitly checks it whereas cloudflare doesnt.

**Review by emilioalvap — APPROVED** [review]
LGTM

**Comment by kvesthy** [comment]
@vigneshshanmugam There is discrepancy about the script below **mfa.token**

const token = mfa.token(params.MFA_GH_SECRET) 

It should be 

const token = **mfa.totp(**params.MFA_GH_SECRET)

Can you please review?

**Comment by vigneshshanmugam** [comment]
@kvesthy Good catch, Sent out a fix to the docs. Thank you.

**Pull request merged**